From: Boyuan Yang <byang@debian.org>
Date: Fri, 15 Aug 2025 20:21:29 -0400
Subject: CMakeLists.txt: Install missing yuvconstants binary

Forwarded: not-needed
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 660003f..9db2aaa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -248,6 +248,7 @@ endif()
 
 # install the conversion tool, .so, .a, and all the header files
 install ( TARGETS yuvconvert DESTINATION ${CMAKE_INSTALL_BINDIR} )
+install ( TARGETS yuvconstants DESTINATION ${CMAKE_INSTALL_BINDIR} )
 install ( TARGETS ${ly_lib_static}						DESTINATION ${CMAKE_INSTALL_LIBDIR} )
 install ( TARGETS ${ly_lib_shared} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} RUNTIME DESTINATION bin ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} )
 install ( DIRECTORY ${PROJECT_SOURCE_DIR}/include/		DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} )
